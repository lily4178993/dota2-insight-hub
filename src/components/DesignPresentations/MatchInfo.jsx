import React from 'react';
import PropTypes from 'prop-types';
import matchesImage from '../../assets/trophy.png';
import './matchInfo.css';

const MatchInfo = ({ data }) => (
  <div className="match-info">
    <div className="match-image">
      <img src={matchesImage} alt="trophy icon" />
      <p>
        <strong>ID:</strong>
        {' '}
        {data.match_id}
      </p>
    </div>
    <div>
      <h2>Match Details</h2>
      <ul>
        <li>
          <strong>Duration:</strong>
          {' '}
          {data.duration}
          {' '}
          seconds
        </li>
        <li>
          <strong>Start Time:</strong>
          {' '}
          {new Date(data.start_time * 1000).toLocaleString()}
        </li>
        <li>
          <strong>League:</strong>
          {' '}
          {data.league_name}
        </li>
        <li>
          <strong>Radiant Team:</strong>
          {' '}
          {data.radiant_name}
        </li>
        <li>
          <strong>Dire Team:</strong>
          {' '}
          {data.dire_name}
        </li>
        <li>
          <strong>Radiant Score:</strong>
          {' '}
          {data.radiant_score}
        </li>
        <li>
          <strong>Dire Score:</strong>
          {' '}
          {data.dire_score}
        </li>
      </ul>
    </div>
  </div>
);

MatchInfo.propTypes = {
  data: PropTypes.shape({
    match_id: PropTypes.number.isRequired,
    duration: PropTypes.number.isRequired,
    start_time: PropTypes.number.isRequired,
    league_name: PropTypes.string.isRequired,
    radiant_name: PropTypes.string.isRequired,
    dire_name: PropTypes.string.isRequired,
    radiant_score: PropTypes.number.isRequired,
    dire_score: PropTypes.number.isRequired,
  }).isRequired,
};

export default MatchInfo;
