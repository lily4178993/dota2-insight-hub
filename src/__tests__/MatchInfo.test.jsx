import React from 'react';
import { render } from '@testing-library/react';
import MatchInfo from '../components/DesignPresentations/MatchInfo';

const sampleData = {
  match_id: 12345,
  duration: 1800,
  start_time: 1634119200,
  league_name: 'Sample League',
  radiant_name: 'Radiant Team',
  dire_name: 'Dire Team',
  radiant_score: 25,
  dire_score: 20,
};

test('MatchInfo component matches snapshot', () => {
  const { asFragment } = render(<MatchInfo data={sampleData} />);
  expect(asFragment()).toMatchSnapshot();
});
