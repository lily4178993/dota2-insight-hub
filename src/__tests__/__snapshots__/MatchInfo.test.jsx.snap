// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchInfo component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="match-info"
  >
    <div
      class="match-image"
    >
      <img
        alt="trophy icon"
        src="trophy.png"
      />
      <p>
        <strong>
          ID:
        </strong>
         12345
      </p>
    </div>
    <div>
      <h2>
        Match Details
      </h2>
      <ul>
        <li>
          <strong>
            Duration:
          </strong>
           1800 seconds
        </li>
        <li>
          <strong>
            Start Time:
          </strong>
           13/10/2021 10:00:00
        </li>
        <li>
          <strong>
            League:
          </strong>
           Sample League
        </li>
        <li>
          <strong>
            Radiant Team:
          </strong>
           Radiant Team
        </li>
        <li>
          <strong>
            Dire Team:
          </strong>
           Dire Team
        </li>
        <li>
          <strong>
            Radiant Score:
          </strong>
           25
        </li>
        <li>
          <strong>
            Dire Score:
          </strong>
           20
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
